packages/devtools_app/lib/src/provider/provider_screen.dart

```diff
@@ -0,0 +1,805 @@
+    final backgroundColor =
+        isSelected ? colorScheme.selectedRowBackgroundColor : null;
+
+    return GestureDetector(
+      behavior: HitTestBehavior.opaque,
+      onTap: () => context.read(_selectedProviderIdProvider).state = providerId,
+      child: Container(
+        color: backgroundColor,
+        padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
+        child: state.when(
+          loading: () => const CenteredCircularProgressIndicator(),
+          error: (err, stack) => Text('<Failed to load> $err\n\n$stack'),
+          data: (node) {
+            return Text('${node.type}()');
+          },
+        ),
+      ),
+    );
+  }
+}
+
+/// A delegate that allows using ListView with an undetermined list length
+/// while preserve the "build only what is visible" behaviour.
+class SliverIterableChildDelegate extends SliverChildDelegate {
+  SliverIterableChildDelegate(
+    this.children, {
+    this.estimatedChildCount,
+  });
+
+  final Iterable<Widget> children;
+  int _lastAccessedIndex;
+  Iterator<Widget> _lastAccessedIterator;
+
+  @override
+  Widget build(BuildContext context, int index) {
+    if (_lastAccessedIndex == null || _lastAccessedIndex > index) {
+      _lastAccessedIndex = -1;
+      _lastAccessedIterator = children.iterator;
+    }
+
+    while (_lastAccessedIndex < index) {
+      _lastAccessedIterator.moveNext();
+      _lastAccessedIndex++;
+    }
+
+    return _lastAccessedIterator.current;
+  }
+
+  @override
+  final int estimatedChildCount;
+
+  @override
+  bool shouldRebuild(SliverIterableChildDelegate oldDelegate) {
+    return children != oldDelegate.children ||
+        _lastAccessedIndex != oldDelegate._lastAccessedIndex ||
+        _lastAccessedIterator != oldDelegate._lastAccessedIterator;
+  }
+}
+
+class _ProviderValue extends ConsumerWidget {
+  const _ProviderValue({Key key}) : super(key: key);
+
+  Iterable<Widget> _buildError(Object error, StackTrace stackTrace) {
+    if (error is SentinelException) {
+      return [Text(error.sentinel.valueAsString)];
+    }
+
+    log(error, LogLevel.error);
+    if (stackTrace != null) {
+      log(stackTrace);
+    }
+
+    return const [Text('<unknown error>')];
+  }
+
+  Iterable<Widget> _buildListViewItems(
+    BuildContext context,
+    ScopedReader watch, {
+    @required InstancePath path,
+    bool disableExpand = false,
+  }) {
+    // TODO(rrousselGit): update riverpod to not prevent `watch` inside listview.builder
+    return watch(instanceProvider(path)).when(
+      // TODO: during loading, return the previous result to avoid flickers
+      loading: () => const [Text('loading...')],
+      error: _buildError,
+      data: (instance) sync* {
+        final isExpanded = watch(_isExpandedProvider(path));
+        yield _buildHeader(
+          instance,
+          isExpanded: isExpanded,
+          disableExpand: disableExpand,
+        );
+
+        if (isExpanded.state) {
```

Comment:
this seems like it reimplementing the value display for the Debugger view. Lets see if we can integrate this functionality into the general debugger view instead or layer it on top of the debugger view which already deals with list, map, set, etc.

Reply:
The existing value display in the debugger view is fairly basic at the moment. I'd be thrilled to see it include some of the functionality it looks like you have added here.

Reply:
You mean the `Variables` panel inside the Debugger tab?